### PR TITLE
bugfix/2.2.13/discoveries non transactional

### PR DESCRIPTION
**Pull request by aliothe**
All discoveries should be non transactional.

### CODE REVIEW

File: casual/casual-caller/src/test/groovy/se/laz/casual/connection/caller/CasualCallerImplTest.groovy [hunk 1]
```diff
@@ -286,27 +275,25 @@ class CasualCallerImplTest extends Specification
...
+       TransactionLess transactionLess = Mock(TransactionLess) {
+          0 * tpcall(_)
        }
-       def caller = new CasualCallerImpl(lookup, connectionFactoryProvider, new TransactionLess(transactionManagerProvider))
+       def caller = new CasualCallerImpl(lookup, connectionFactoryProvider, transactionLess)
        caller.tpCaller = Mock(TpCallerFailover)
-       1 * transactionManager.resume(_)
        when:
-       caller.tpcall("foo", Mock(CasualBuffer), Flag.of(AtmiFlags.TPNOTRAN))
+       caller.tpcall("foo", Mock(CasualBuffer), Flag.of(AtmiFlags.NOFLAG))
```
Comment (TobiasLeo) [inline casual/casual-caller/src/test/groovy/se/laz/casual/connection/caller/CasualCallerImplTest.groovy:284]: Why send NOFLAG in this test and expect 0 tpcalls in transactionLess? Seems contrary to test description where I would expect TPNOTRAN flag is sent and tpCall is invoked for the call in transactionLess

Comment (aliothe) [reply casual/casual-caller/src/test/groovy/se/laz/casual/connection/caller/CasualCallerImplTest.groovy:284]: No flag then it is in transaction.
We only check if TPNOTRAN is sent to us via that API, if it is - then it should be transactionless tp(a)call.

Comment (aliothe) [reply casual/casual-caller/src/test/groovy/se/laz/casual/connection/caller/CasualCallerImplTest.groovy:284]: It was wrong before.
With CasualCaller::tpcall(AtmiFlags.NOFLAG) - it is transactional -> TransactionLess is not used.
It is only used if you provide TPNOTRAN.

Comment (TobiasLeo) [reply casual/casual-caller/src/test/groovy/se/laz/casual/connection/caller/CasualCallerImplTest.groovy:284]: Yes, but the test probably shouldn't be called 'TPNOTRAN tpcall, in transaction' if the tpcall is done with no flags. Just 'tpcall, in transaction' maybe?

**Review by TobiasLeo — APPROVED** [review]

**Pull request merged**